elastic/kibana
# [OneDiscover][Extension] DataTable Row Actions

- Closes https://github.com/elastic/kibana/issues/186637
- Closes https://github.com/elastic/kibana/issues/186808

## Summary

- [x] Extend UnifiedDataTable with a new `rowAdditionalLeadingControls` prop to render additional leading controls
- [x] In case of many actions, collapse the rest of them under "More" button
- [x] New OneDiscover extension
- [x] Convert from `customControlColumnsConfiguration` to the new prop. Refactor actions format in Log Explorer.
- [x] Swap the default "select" and "expand" control columns ~if custom row actions are specified~
- [x] Add to example OneDiscover profile
- [x] Add functional and units tests

<img width="858" alt="Screenshot 2024-07-26 at 16 00 17" src="https://github.com/user-attachments/assets/68832abc-a498-4ec6-8333-79ad5f83855b">
<img width="1139" alt="Screenshot 2024-07-26 at 16 00 47" src="https://github.com/user-attachments/assets/8ff99ac1-21b0-4687-b548-fbf3c5517808">

### Testing 

For testing the example profile:
- add `discover.experimental.enabledProfiles: ['example-root-profile', 'example-data-source-profile', 'example-document-profile']` to kibana.dev.yml 
- start kibana
- make sure to have an index with `my-example-logs` name or create an alias to an existing index:
```
POST _aliases
{
  "actions": [
    {
      "add": {
        "index": "kibana_sample_data_logs",
        "alias": "my-example-logs"
      }
    }
  ]
}
```
- create a data view for `my-example-logs` index


### Follow up for Security solution

The following items would require deprecation/refactoring in components on Security Solution pages to have consistent UX (can result in 500+ lines of code changes):
- Convert from `externalControlColumns` to the new prop `rowAdditionalLeadingControls`
- ~Convert from `trailingControlColumns` to a normal column. `trailingControlColumns` is deprecated~
  - https://github.com/elastic/kibana/issues/189294
- Use `getRowIndicator` prop on UnifiedDataTable instead of `border-left` style
  - https://github.com/elastic/kibana/issues/189295


### Checklist
